integration-tests/src/test/java/com/linkedin/kafka/clients/producer/RoundRobinPartitioner.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.producer;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.clients.producer.Partitioner;
+import org.apache.kafka.common.Cluster;
+import org.apache.kafka.common.PartitionInfo;
+
+
+/**
+ * a partitioner that sends every msg to a different partition, round-robin style
+ */
+public class RoundRobinPartitioner implements Partitioner {
+  private int counter = 0;
```

Comment:
Since producers are multi-thread compliant, Should this be AtomicInteger?

Reply:
in general yes, but the particular test i have is single threaded.